### PR TITLE
Update to last versions of some very infrequently updated repos

### PR DESCRIPTION
**Pull request by dims**
Some repos we use have not been updated in a while. BUT we are not on the latest sha/tag for them either. Updating to them will help those of us who keep looking for delta(s) in dependencies and whether there are any fixes that need to be picked up.

These repos in particular definitely seem like they won't get an updates anytime soon! 

`Last updated Sep 28, 2017` - https://github.com/mohae/deepcopy/compare/491d3605edfb...c48cc78d4826
`Last updated Feb 14, 2019` - https://github.com/armon/circbuf/compare/bbbad097214e...5111143e8da2
`Last updated Jun 11, 2019` - https://github.com/gregjones/httpcache/compare/9cad4c3443a7...901d90724c79
`Last updated Apr  7, 2021` - https://github.com/exponent-io/jsonpath/compare/d6023ce2651d...1de76d718b3f
`Last updated Nov 25, 2022` - https://github.com/xiang90/probing/compare/43a291ad63a2...a49e3df8f510
~~WONT-FIX `Last updated Mar  1, 2023` - https://github.com/asaskevich/govalidator/compare/f61b66f89f4a...a9d515a09cc2 WONT-FIX~~
`Last updated Mar 27, 2023` - https://github.com/stoewer/go-strcase/compare/v1.2.0...v1.3.0
`Last updated Apr  1, 2023` - https://github.com/jonboulle/clockwork/compare/v0.2.2...v0.4.0

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: vendor/github.com/asaskevich/govalidator/patterns.go [hunk 1]
```diff
@@ -4,49 +4,55 @@ import "regexp"
 
 // Basic regular expressions for validating strings
 const (
```
Comment (liggitt) [inline vendor/github.com/asaskevich/govalidator/patterns.go:6]: these updates are problematic, they change the API validation of CRD format validators

@jpbetz @sttts @deads2k  we should isolate use of these in kube-openapi and stop depending on an external library for this API 
surface

File: go.mod [hunk 2]
```diff
@@ -128,7 +128,7 @@ require (
 	github.com/MakeNowJust/heredoc v1.0.0 // indirect
 	github.com/NYTimes/gziphandler v1.1.1 // indirect
 	github.com/antlr4-go/antlr/v4 v4.13.0 // indirect
-	github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a // indirect
+	github.com/asaskevich/govalidator v0.0.0-20230301143203-a9d515a09cc2 // indirect
```
Comment (liggitt) [inline go.mod:131]: revert the govalidator change, it modifies CRD format validation

File: vendor/github.com/stoewer/go-strcase/camel.go [hunk 3]
```diff
@@ -27,6 +27,9 @@ func camelCase(s string, upper bool) string {
 				buffer = append(buffer, toUpper(curr))
 			} else if isLower(prev) {
 				buffer = append(buffer, curr)
+			} else if isUpper(prev) && isUpper(curr) && isLower(next) {
+				// Assume a case like "R" for "XRequestId"
+				buffer = append(buffer, curr)
```
Comment (liggitt) [inline vendor/github.com/stoewer/go-strcase/camel.go:32]: looks like this is only used by CEL... not sure the impact of this change, if any

https://github.com/kubernetes/kubernetes/blob/429edc5f25efb76743677217326dc19360ab5c2f/vendor/github.com/google/cel-go/common/types/map.go#L206-L207

cc @jpbetz for input on whether this is ok

**Comment by dims** [comment]
Ack, will do ( `revert the govalidator change` )

File: go.mod (hunk 2 above)
Comment (dims) [reply go.mod:131]: on it

**Comment by liggitt** [comment]
https://github.com/kubernetes/kubernetes/pull/128064#discussion_r1800164315 is the only open question, everything else lgtm

**Comment by dims** [comment]
/assign @saschagrunert 

(for cri-client)

File: vendor/github.com/stoewer/go-strcase/camel.go (hunk 3 above)
Comment (jpbetz) [reply vendor/github.com/stoewer/go-strcase/camel.go:32]: We do not use this in Kubernetes.

We [use our own types](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/apiserver/pkg/cel/common) for CRDs, and we use `stringIfaceMapAccessor` in DRA.

cc @TristonianJones for potential impact to other CEL users.

Comment (dims) [reply vendor/github.com/stoewer/go-strcase/camel.go:32]: thanks for confirming @jpbetz !

**Comment by pacoxu** [comment]
/retest
for scheduling timeout of CI pod

File: vendor/github.com/stoewer/go-strcase/camel.go (hunk 3 above)
Comment (TristonianJones) [reply vendor/github.com/stoewer/go-strcase/camel.go:32]: @dims This change shouldn't impact Kubernetes. The code in question is used to try to map from a CEL Map value to a Golang struct during calls to `ConvertToNative` which I don't think are in the K8s call path, or at least won't be used for those purposes. This should be safe to update.

Comment (dims) [reply vendor/github.com/stoewer/go-strcase/camel.go:32]: @TristonianJones thanks for confirming!

**Pull request merged**